File: backend/src/main/java/br/ufpr/sept/so2/modules/solicitacoes/infrastructure/persistence/SolicitacaoEventoJpaRepository.java
package br.ufpr.sept.so2.modules.solicitacoes.infrastructure.persistence;

import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.UUID;

public interface SolicitacaoEventoJpaRepository extends JpaRepository<SolicitacaoEventoJpaEntity, UUID> {

    List<SolicitacaoEventoJpaEntity> findBySolicitacaoIdOrderByCreatedAtDesc(UUID solicitacaoId);
}

File: backend/src/main/kotlin/br/ufpr/sept/so2/modules/solicitacoes/infrastructure/persistence/SolicitacaoEventoJpaRepository.kt
package br.ufpr.sept.so2.modules.solicitacoes.infrastructure.persistence

import org.springframework.data.jpa.repository.JpaRepository
import java.util.UUID

interface SolicitacaoEventoJpaRepository : JpaRepository<SolicitacaoEventoJpaEntity, UUID> {

    fun findBySolicitacaoIdOrderByCreatedAtDesc(solicitacaoId: UUID): List<SolicitacaoEventoJpaEntity>
}
